lib/datapackage.py

```diff
@@ -0,0 +1,394 @@
+from __future__ import print_function
+from builtins import input
+import os
+import json
+from time import sleep
+from retriever import SCRIPT_LIST, HOME_DIR
+
+short_names = [script.shortname.lower() for script in SCRIPT_LIST()]
+
+
+def is_empty(val):
+    """Check if a variable is an empty string or an empty list"""
+    return val == "" or val == []
+
+
+def clean_input(prompt="", split_char='', ignore_empty=False, dtype=None):
+    """Clean the user-input from the CLI before adding it"""
+    while True:
+        val = input(prompt).strip()
+        # split to list type if split_char specified
+        if split_char != "":
+            val = [v.strip() for v in val.split(split_char) if v.strip() != ""]
+        # do not ignore empty input if not allowed
+        if not ignore_empty and is_empty(val):
+            print("\tError: empty input. Need one or more values.\n")
+            continue
+        # ensure correct input datatype if specified
+        if not is_empty(val) and dtype != None:
```

Comment:
would this check be fine for all cases or could we use `dtype != None ` to dtype is Not None